influxdb3/tests/server/write.rs

```diff
@@ -0,0 +1,268 @@
+use hyper::StatusCode;
+use pretty_assertions::assert_eq;
+
+use crate::TestServer;
+
+#[tokio::test]
+async fn api_v1_write_request_parsing() {
+    let server = TestServer::spawn().await;
+    let client = reqwest::Client::new();
+    let write_url = format!("{base}/write", base = server.client_addr());
+    let write_body = "cpu,host=a usage=0.5";
+
+    #[derive(Debug)]
+    struct TestCase {
+        db: Option<&'static str>,
+        precision: Option<&'static str>,
+        rp: Option<&'static str>,
+        expected: StatusCode,
+    }
+
+    let test_cases = [
+        TestCase {
+            db: None,
+            precision: None,
+            rp: None,
+            expected: StatusCode::BAD_REQUEST,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: None,
+            rp: None,
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: Some("s"),
+            rp: None,
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: Some("ms"),
+            rp: None,
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: Some("us"),
+            rp: None,
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: Some("ns"),
+            rp: None,
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: Some("invalid"),
+            rp: None,
+            expected: StatusCode::BAD_REQUEST,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: None,
+            rp: Some("bar"),
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: None,
+            rp: Some("default"),
+            expected: StatusCode::OK,
+        },
+        TestCase {
+            db: Some("foo"),
+            precision: None,
+            rp: Some("autogen"),
+            expected: StatusCode::OK,
+        },
+    ];
+
+    for t in test_cases {
```

Comment:
I like this design of a list of test cases to iterate through. Very nice. We might be able to bring this pattern to other tests we have.